grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy

```diff
@@ -55,48 +48,61 @@ class SubmissionController extends AuthenticatedController {
             }
         }
         SubmissionType defaultType = SubmissionType.getForLabel(params.get("type").toString()) ?: SubmissionType.Amendment.label
-        render(view: '/submission/submission',
-                model: [issue:                      issue,
-                        submission:                 submission,
-                        docTypes:                   PROJECT_DOC_TYPES,
-                        submissionTypes:            submissionTypes,
-                        submissionNumberMaximums:   submissionNumberMaximums,
-                        defaultType:                defaultType
-                ])
+        render( [issue:                      issue,
+                 typeLabel:                  issue.typeLabel,
+                 submission:                 submission,
+                 documents:                  submission?.documents,
+                 docTypes:                   PROJECT_DOC_TYPES,
+                 submissionTypes:            submissionTypes,
+                 submissionNumberMaximums:   submissionNumberMaximums,
+                 defaultType:                defaultType
+        ] as JSON)
     }
 
     def save() {
-        Issue issue = queryService.findByKey(params.projectKey)
-        Submission submission
-        if (params?.submissionId) {
-            submission = Submission.findById(params?.submissionId)
-            submission.number = Integer.parseInt(params?.submissionNumber)
-            submission.type = params?.submissionType
-            submission.comments = params?.comments
-            if (!submission.save(flush: true)) {
-                flash.message = submission.errors.allErrors.collect { it }.join("<br/>")
+        JsonParser parser = new JsonParser()
+        JsonArray dataSubmission = parser.parse(request.parameterMap["submission"].toString())
```

Comment:
`parser.parse` returns a `JsonElement`, not a `JsonArray`, so you need to cast it. Something along these lines:
```
        JsonElement el = parser.parse(request.parameterMap["submission"].toString())
        if (el.jsonArray) {
            JsonArray dataSubmission = el.asJsonArray
        }
```